activemq-broker/src/main/java/org/apache/activemq/broker/region/Queue.java

```diff
@@ -2100,7 +2100,7 @@ private PendingList doPageInForDispatch(boolean force, boolean processExpired, i
                                 ConnectionContext connectionContext = createConnectionContext();
                                 dropMessage(ref);
                                 if (gotToTheStore(ref.getMessage())) {
-                                    LOG.debug("Duplicate message {} from cursor, removing from store", this, ref.getMessage());
+                                    LOG.debug("{}, duplicate message {} from cursor, removing from store", this, ref.getMessage());
```

Comment:
Can you please use:

```
LOG.debug("Duplicate message {] from cursor, removing from store", ref.getMessage());
```

`this` is useless here.

Reply:
Right, the problem was, that `this` was used as first parameter in the log format string that had only one parameter. So the second  parameter `ref.getMessage()` was silently dropped.
As I didn't know, if `this` was needed for your analysis of the log messages, I added a second parameter placeholder.
I will try to update this PR.